### PR TITLE
Only query single route if segmentRange specified

### LINKED ISSUE #520: connect links outside date range do not load
Just shows the drive list

### PR DESCRIPTION
**Pull request by vishalkrishnads**

### CODE REVIEW

**Comment by sshane** [comment]
Loading all routes isn't an appropriate solution (speed), we should go and check how this used to work and replicate that. Was it an API query using the time ranges? Perhaps we can just query w/ the route name

**Comment by adeebshihadeh** [comment]
> Loading all routes isn't an appropriate solution (speed), we should go and check how this used to work and replicate that. Was it an API query using the time ranges? Perhaps we can just query w/ the route name

We used to have the time range in the URL. We'll have to update the endpoint to also take a route name I guess.

**Comment by vishalkrishnads** [comment]
Yeah that makes the most sense.

**Comment by sshane** [comment]
Does this have everything we need? https://github.com/commaai/comma-api/blob/d23b152ea64bfcd0de0937d2d525d556870820b6/src/drives.js#L17

**Comment by vishalkrishnads** [comment]
Sorry, this doesn't. We need `segment_start_times`, `segment_end_times` & `segment_numbers` in the response to make this work. I think this should wait until #536 is closed.

**Comment by vishalkrishnads** [comment]
With this, I'm getting a CORS error. Can't seem to get around it. Any idea why?

cors

I also tried with a plain JS `fetch()` without using the client JS library. Doesn't seem to work either, throws the same error.  I'm sure it's something I'm doing wrong because this same endpoint is [used](https://github.com/commaai/new-connect/blob/28699ee8b517b0607dd96fe81277e2170ef62121/src/api/route.ts#L36) in the new connect. But I can't figure it out.

**Comment by vishalkrishnads** [comment]
This issue has been fixed. for some reason, none of the CF pages related to my PR's are updating on subsequent commits. So, the URL related to this PR is outdated. However, the issue has been solved and the drives are loading.

File: src/actions/index.js [hunk 1]
```diff
@@ -53,14 +54,24 @@ export function checkRoutesData() {
         return;
       }
 
-      const routes = routesData.map((r) => {
+      if (!state.currentRoute && state.segmentRange) {
```
Comment (sshane) [inline src/actions/index.js]: Does this unnecessarily call the API when you click on a route inside the time range?

Comment (sshane) [reply src/actions/index.js]: Also, a single line comment explaining this would be a good idea!

Comment (vishalkrishnads) [reply src/actions/index.js]: Sorry for overlooking this! I've added a comment and fixed it.

**Comment by sshane** [comment]
This still doesn't work properly... It never recovers from this state:

image

Have a PR to properly fix this: https://github.com/commaai/connect/pull/544

**Comment by vishalkrishnads** [comment]
i can't see the issue. seems fine for me

https://github.com/user-attachments/assets/d7e0eb1f-9476-4c7c-81e7-c297d3237f28

**Comment by sshane** [comment]
Try loading a route outside the first 10 that load by default, you may need to zoom in

**Comment by sshane** [comment]
Very strange, I can reproduce it locally with `pnpm start` but not the deployed preview. Maybe something different in how it's run in production.

I changed it to only query the specific route on loading a route from URL, makes a bit more sense

**Comment by vishalkrishnads** [comment]
seems like it works now 😅

are you going to merge this or do you want me to clean up anything more?

**Pull request merged**